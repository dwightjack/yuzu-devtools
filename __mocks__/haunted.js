const haunted = require('haunted/index');

module.exports = {
  ...haunted,
  useContext: jest.fn(),
};
