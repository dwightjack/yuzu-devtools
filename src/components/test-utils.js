export const mockComponent = (tag) => (props) => {
  const el = document.createElement(tag);
  Object.keys(props).forEach((key) => {
    el.setAttribute(key, props[key]);
  });
  return el;
};
