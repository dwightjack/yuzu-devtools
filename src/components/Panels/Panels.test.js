import Split from 'split-grid';
import { virtual } from 'haunted';
import { toHTMLAsync } from '../test-utils';

const Panels = virtual(require('./Panels').default);

jest.mock('split-grid', () => jest.fn(() => ''));

describe('Panels', () => {
  test('matches default snapshot', async () => {
    const { html } = await toHTMLAsync(Panels);
    expect(html).toMatchSnapshot();
  });

  test('Calls module "Split" on connected', async () => {
    await toHTMLAsync(Panels);
    expect(Split).toHaveBeenCalled();
  });
});
