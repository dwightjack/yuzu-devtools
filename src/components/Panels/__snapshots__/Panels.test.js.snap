// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panels matches default snapshot 1`] = `
<div class="root">
  <slot name="main">
  </slot>
  <div class="gutter">
  </div>
  <slot name="side">
  </slot>
  <div>
  </div>
</div>
`;
